File: tests/fixtures/sample.go
// Go fixture for symbol extraction tests
package main

type Config struct {
    Host string
    Port int
}

func main() {
    println("hello")
}

func Add(a int, b int) int {
    return a + b
}

func (c *Config) GetHost() string {
    return c.Host
}

type User struct {
    Name string
}

func (u *User) Greet() string {
    return "Hello, " + u.Name
}

type Calculator struct {
    value int
}

func (c *Calculator) Add(amount int) {
    c.value += amount
}

func (c *Calculator) GetValue() int {
    return c.value
}

File: tests/fixtures/sample.rs
// Rust fixture for symbol extraction tests
pub struct Config {
    host: String,
    port: u16,
}

impl Config {
    pub fn new(host: &str, port: u16) -> Self {
        Config { host: host.to_string(), port }
    }

    pub fn get_host(&self) -> &str {
        &self.host
    }
}

fn add(a: i32, b: i32) -> i32 {
    a + b
}

struct Calculator {
    value: i32,
}

impl Calculator {
    fn new() -> Self {
        Calculator { value: 0 }
    }

    fn add(&mut self, amount: i32) {
        self.value += amount;
    }

    fn get_value(&self) -> i32 {
        self.value
    }
}

pub fn main() {
    println!("Hello");
}